docs/development/when_to_rebase.rst

```diff
@@ -0,0 +1,45 @@
+*********************************
+When to rebase and squash commits
+*********************************
+
+This page describes recommendations for when to rebase pull requests and when to
+combine/squash commits.
+
+When to remove or combine/squash commits
+========================================
+
+Pull requests **must** be rebased and at least partially squashed (but not
+necessarily squashed to a single commit) if large (approximately >10KB)
+non-source code files (e.g. images, data files, etc.) are added and then removed
+or modified in the PR commit history (The squashing should remove all but the
+last addition of the file to not use extra space in the repository).
+
+Combining/squashing commits is **encouraged** but not required where the number
+of commits is excessive for the changes made.  The definition of 'excessive' is
+subjective, but would never apply for 5 or fewer commits even for a trivial
```

Comment:
I think more commits than lines changed is "excessive" even if less than 5.

Reply:
I would suggest here to be less explicit, and just write
```
Combining/squashing commits is **encouraged** when the number of commits
is excessive for the changes made. The definition of 'excessive' is of course
subjective, but in general one should attempt to have individual commits be
units of change, and not include reversions.

As a concrete example, for a change affecting < 10 lines of source code and
including a changelog entry, more than a few commits would be excessive.
For a larger pull request adding significant functionality, however, more
commits may well be appropriate.
```

Reply:
Looking at this again, I'd still like my change to be incorporated...